client/pages/search-results.jsx

```diff
@@ -0,0 +1,132 @@
+import React from 'react';
+import parseRoute from '../lib/parse-route';
+
+export default class SearchResults extends React.Component {
+  constructor(props) {
+    super(props);
+    const startValue = this.formatSearchResults();
+    this.state = {
+      searchValue: startValue,
+      isLoading: true,
+      results: []
+    };
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+    this.fetchSearchResults = this.fetchSearchResults.bind(this);
+  }
+
+  handleInputChange(event) {
+    this.setState({ searchValue: event.target.value });
+  }
+
+  formatSearchResults() {
+    const paramString = this.props.params.toString();
+    const query = paramString.slice(2)
+      .replaceAll('+', ' ')
+      .replaceAll('%27', '\'')
+      .replaceAll('%21', '!')
+      .replaceAll('%2C', ',')
+      .replaceAll('%25', '%')
+      .replaceAll('%40', '@')
+      .replaceAll('%23', '#')
+      .replaceAll('%24', '$')
+      .replaceAll('%28', '(')
+      .replaceAll('%29', ')')
+      .replaceAll('%3D', '=');
```

Comment:
I could be wrong, but does `decodeURIComponent(paramString.slice(2))` not work here?